### PR TITLE
cpp object detection demo: ssh-mxnet support

### PR DESCRIPTION
**Pull request by ivikhrev**
Added support for ssh-mxnet model.

### CODE REVIEW

File: demos/common/cpp/models/src/detection_model_retinaface.cpp [hunk 1]
```diff
@@ -336,22 +339,23 @@ std::unique_ptr<ResultBase> ModelRetinaFace::postprocess(InferenceResult& infRes
...
+            filterLandmarks(landmarks, validIndices, landmarksRaw, anchorNum, anchors[idx], landmarkStd);
+        }
         if (shouldDetectMasks) {
             const auto masksRaw = infResult.outputsData[separateOutputsNames[OT_MASKSCORES][idx]];
             filterMasksScores(masks, validIndices, masksRaw, anchorNum);
         }
     }
     // --------------------------- Apply Non-maximum Suppression ----------------------------------------------------------
-    auto keep = nms(bboxes, scores, boxIOUThreshold);
+    auto keep = nms(bboxes, scores, boxIOUThreshold, !shouldDetectLandmarks);
```
Comment (Wovchena) [inline demos/common/cpp/models/src/detection_model_retinaface.cpp:358]: Why does nms depend on landmarks presence? Is it just a way of distinguishing `ssh-mxnet` and `retinaface` and procedures they had for training? If so, please add a comment explaining that in code.

File: demos/object_detection_demo/cpp/main.cpp [hunk 2]
```diff
@@ -67,7 +67,7 @@ static const char num_streams_message[] = "Optional. Number of streams to use fo
 "<device1>:<nstreams1>,<device2>:<nstreams2> or just <nstreams>)";
 static const char no_show_processed_video[] = "Optional. Do not show processed video.";
 static const char utilization_monitors_message[] = "Optional. List of monitors to show initially.";
-static const char iou_thresh_output_message[] = "Optional. Filtering intersection over union threshold for overlapping boxes (YOLOv3 only).";
+static const char iou_thresh_output_message[] = "Optional. Filtering intersection over union threshold for overlapping boxes.";
```
Comment (Wovchena) [inline demos/object_detection_demo/cpp/main.cpp:70]: It looks `showUsage()` doesn't print `iou_thresh_output_message` at all. Please update it and readme.

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: demos/object_detection_demo/cpp/main.cpp (hunk 2 above)
Comment (ivikhrev) [reply demos/object_detection_demo/cpp/main.cpp:70]: Updated

File: demos/common/cpp/models/src/detection_model_retinaface.cpp (hunk 1 above)
Comment (ivikhrev) [reply demos/common/cpp/models/src/detection_model_retinaface.cpp]: In `nms` function for ssh-mxnet we should include boundaries in areas calculations, so, yes,  by passing `!shouldDetectLandmarks` we distinguish it from retinaface models .

File: demos/common/cpp/models/src/detection_model_retinaface.cpp [hunk 3]
```diff
@@ -336,22 +339,24 @@ std::unique_ptr<ResultBase> ModelRetinaFace::postprocess(InferenceResult& infRes
...
+        }
         if (shouldDetectMasks) {
             const auto masksRaw = infResult.outputsData[separateOutputsNames[OT_MASKSCORES][idx]];
             filterMasksScores(masks, validIndices, masksRaw, anchorNum);
         }
     }
     // --------------------------- Apply Non-maximum Suppression ----------------------------------------------------------
-    auto keep = nms(bboxes, scores, boxIOUThreshold);
+    // !shouldDetectLandmarks determines nms behavior, if true - boundaries are included in areas calculation for ssh-mxnet model
+    auto keep = nms(bboxes, scores, boxIOUThreshold, !shouldDetectLandmarks);\
```
Comment (Wovchena) [inline demos/common/cpp/models/src/detection_model_retinaface.cpp:359]:
```suggestion
    auto keep = nms(bboxes, scores, boxIOUThreshold, !shouldDetectLandmarks);
```

Comment (ivikhrev) [reply demos/common/cpp/models/src/detection_model_retinaface.cpp:359]: oops, didn't notice it

**Review by Wovchena — APPROVED** [review]

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Pull request merged**